pytest_postgresql/factories/process.py

```
@@ -124,62 +124,67 @@ def postgresql_proc_fixture(
+                        )
+                    used_ports.add(pg_port)
+                    with (port_filename_path).open("x") as port_file:
+                        port_file.write(f"pg_port {pg_port}\n")
+                    break
+                except FileExistsError:
+                    if n >= config.port_search_count:
+                        raise PortForException(
+                            f"Attempted {n} times to select ports. "
+                            f"All attempted ports: {', '.join(map(str, used_ports))} are already "
+                            f"in use, probably by other instances of the test."
+                        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)